api/METRICS.md

```diff
@@ -35,72 +45,260 @@ acc_ratio = accuracy_score_ratio(y_true, y_pred, sensitive_features=sf, **other_
 |`false_positive_rate` | G,D,R | class | | &#x2713; |
-|`false_negative_rate` | G | class | | &#x2713; |
+|`false_negative_rate` | G,D,R | class | | &#x2713; |
 |`true_positive_rate` | G,D,R | class | | &#x2713; |
-|`true_negative_rate` | G | class | | &#x2713; |
-|`equalized_odds` | D,R | class | max of difference or ratio under `true_positive_rate`, `false_positive_rate` | - |
-|`precision_score`| G | class | sklearn | &#x2713; |
-|`recall_score`| G | class | sklearn | &#x2713; |
-|`f1_score`| G | class | sklearn | - |
-|`roc_auc_score`| G | prob | sklearn | - |
-|`log_loss`| G | prob | sklearn | - |
-|`mean_squared_error`| G | prob, reg | sklearn | - |
-|`r2_score`| G | reg | sklearn | - |
+|`true_negative_rate` | G,D,R | class | | &#x2713; |
+|`selection_rate`| G,D,R | class | | &#x2713; |
+|`demographic_parity`| D,R | class | `selection_rate_difference`, `selection_rate_ratio` | `statistical_parity_difference`, `disparate_impact`|
+|`equalized_odds` | D,R | class | max difference or min ratio under `true_positive_rate`, `false_positive_rate` | - |
+|`accuracy_score`| G,D,R,Min | class | sklearn | `accuracy` |
+|`zero_one_loss` | G,D,R,Max | class | sklearn | `error_rate` |
+|`balanced_accuracy_score` | G,Min | class | sklearn | - |
+|`precision_score`| G,Min | class | sklearn | &#x2713; |
+|`recall_score`| G,Min | class | sklearn | &#x2713; |
+|`f1_score` [TODO]| G,Min | class | sklearn | - |
+|`roc_auc_score`| G,Min | prob | sklearn | - |
+|`log_loss` [TODO]| G,Max | prob | sklearn | - |
+|`mean_absolute_error` | G,Max | reg | sklearn | - |
+|`mean_squared_error`| G,Max | prob, reg | sklearn | - |
+|`r2_score`| G,Min | reg | sklearn | - |
+|`mean_prediction`| G | prob, reg | | - |
+|`_mean_overprediction` | G | class, reg | | - |
+|`_mean_underprediction` | G | class, reg | | - |
+|`_root_mean_squared_error`| G | prob, reg | | - |
+|`_balanced_root_mean_squared_error`| G | prob | | - |
+
+# Harmonizing metrics across modules
+
+Various modules of Fairlearn refer to the related fairness concepts in many different ways. The goal of this part of the proposal is to harmonize their API with the one used in `fairlearn.metrics`.
+
+## Notation
+
+A _metric_ refers to any function that can be evaluated on subsets of the data. We use the notation _metric_(\*) for its value on the whole data set and _metric_(_a_) for its value on the subset of examples with sensitive feature value _a_. We write `<metric>` and `<Metric>` for literals representing the metric name.
+
+For example, if _metric_ is _accuracy_score_, then
+
+* _metric_(\*) = P[_Y=h(X)_]
+* _metric_(_a_) = P[_Y=h(X)_ | _A=a_]
+* `<metric>` = `accuracy_score`
+* `<Metric>` = `AccuracyScore`
+
+## Fairness metrics and fairness constraints
+
+Fairness metrics are expressed in terms of _metric_(\*) and _metric_(_a_). Mitigation algorithms have a notion of _fairness constraints_, expressed in terms of fairness metrics, and a notion of an _objective_, typically equal to _metric_(\*).

```

Comment:
Does the objective have to be in terms of one of the 'raw' metrics (such as accuracy) or could it also be expressed in terms of `accuracy_score_group_min()` etc.?

Reply:
In principle we could extend things as you suggest, but it's not part of this proposal. The proposal only extends from the status quo (which does not even allow specifying the performance metric) with a couple of variants of the form _metric_(\*).